### PR TITLE
EJBCLIENT-254 client fails on Xbootclasspath

### PR DESCRIPTION
**Pull request by n1hility**
DO NOT MERGE YET (Needs update after merged component updates)

https://issues.jboss.org/browse/EJBCLIENT-254
https://issues.jboss.org/browse/JBEAP-12752

Components:

[JBMAR-210](https://issues.jboss.org/browse/JBMAR-210) - https://github.com/jboss-remoting/jboss-marshalling/pull/70
[ELY-1337](https://issues.jboss.org/browse/ELY-1337) - https://github.com/wildfly-security/wildfly-elytron/pull/954

Normal test run does not include bootclasspath runner.

To test with both standard and bootclasspath:
mvn -Dtest-bootclasspath test

Supersedes #300

### CODE REVIEW

**Comment by bstansberry** [comment]
@n1hility I have upgrade all the dependencies.

File: pom.xml [hunk 1]
```diff
@@ -47,7 +47,7 @@
         <!-- 2.1.x also work on JDK9-->
         <version.org.jboss.logging.jboss-logging-tools>2.1.0.Alpha2</version.org.jboss.logging.jboss-logging-tools>
         <version.org.jboss.logmanager>2.0.5.Final</version.org.jboss.logmanager>
-        <version.org.jboss.marshalling>2.0.0.CR1</version.org.jboss.marshalling>
+        <version.org.jboss.marshalling>2.0.1.Final-SNAPSHOT</version.org.jboss.marshalling>
```
Comment (dmlloyd) [inline pom.xml]: Why did you need the snapshot versions?

File: src/main/java/org/jboss/ejb/client/legacy/JBossEJBProperties.java [hunk 2]
```diff
@@ -414,6 +414,10 @@ public static JBossEJBProperties fromProperties(final String fileName, final Pro
         if (stream == null) {
             return null;
         }
+        return fromResource(fileName, stream);
+    }
+
+    private static JBossEJBProperties fromResource(String fileName, InputStream stream) throws IOException {
```
Comment (dmlloyd) [inline src/main/java/org/jboss/ejb/client/legacy/JBossEJBProperties.java]: This all seems OK.

**Comment by dmlloyd** [comment]
@n1hility Can this be rebased and edited to be ready to merge?

**Comment by n1hility** [comment]
replacing this one
